crates/eth-rpc/tests/kkrt_prep.rs

```diff
@@ -0,0 +1,377 @@
+use url::Url;
+
+pub fn get_contract(filename: &str) -> (Abi, EthersBytes) {
+    let path = format!("./tests/compiled_solidity/{filename}");
+    let contents = fs::read_to_string(path).unwrap();
+    let obj: JsonAbi = serde_json::from_str(&contents).unwrap();
+    let JsonAbi::Object(obj) = obj else { panic!() };
+    (serde_json::from_str(&serde_json::to_string(&obj.abi).unwrap()).unwrap(), obj.bytecode.unwrap().to_vec())
+}
+
+pub fn create_call_request(to: Address, selector: [u8; 4]) -> CallRequest {
+    CallRequest {
+        from: Option::None,
+        to: Some(to),
+        gas_price: Option::None,
+        max_fee_per_gas: Option::None,
+        max_priority_fee_per_gas: Option::None,
+        gas: Option::None,
+        value: Option::None,
+        data: Some(selector.into()),
+        nonce: Option::None,
+        chain_id: Option::None,
+        access_list: Option::None,
+        transaction_type: Option::None,
+    }
+}
+
+pub fn create_raw_tx(selector: [u8; 4], eoa_secret: H256, to: Address, args: Vec<U256>, nonce: u64) -> Bytes {
+    // Start with the function selector
+    // Append each argument
+    let mut data: Vec<u8> = selector.to_vec();
+
+    for arg in args {
+        // Ethereum uses big-endian encoding
+        let arg_bytes: [u8; 32] = arg.to_be_bytes();
+        data.extend_from_slice(&arg_bytes);
+    }
+
+    // TODO: dehardcode
+    let chain_id = 1263227476;
+    let gas = 10;
+    // Create a transaction object
+    let transaction = Transaction::Eip1559(TxEip1559 {
+        chain_id,
+        nonce,
+        max_priority_fee_per_gas: Default::default(), // U256 used for large numbers
+
+        max_fee_per_gas: Default::default(), // U256 used for large numbers
+
+        gas_limit: gas, // this may need to be converted to u64
+
+        to: TransactionKind::Call(to),
+
+        value: 0u128, // U256 used for large numbers
+
+        input: data.into(),
+
+        access_list: Default::default(), // empty access list
+    });
+
+    let signature = sign_message(eoa_secret, transaction.signature_hash()).unwrap();
+
+    let signed_transaction = TransactionSigned::from_transaction_and_signature(transaction, signature);
+    let mut raw_tx = BytesMut::new(); // Create a new empty buffer
+
+    signed_transaction.encode_enveloped(&mut raw_tx); // Encode the transaction into the buffer
+
+    raw_tx.to_vec().into()
+}
+
+async fn deploy_kkrt_eth_contract(
+    sequencer_url: Url,
+    deployed_eao_starknet_account: FieldElement,
+    eoa_secret: H256,
+    contract_name: &str,
+) -> Option<Vec<FieldElement>> {
+    // need contract artifact
+    // turns into the data field of a tx_payload
+    // that gets signed and executed in the starknet eoa 'execute'/invoke
+    // can first experiment with doing a call for `get_nonce`
+
+    let signing_key = SigningKey::from_secret_scalar(FieldElement::from_byte_slice_be(eoa_secret.as_slice()).unwrap());
+
+    let eoa_starknet_account = SingleOwnerAccount::new(
```

Comment:
I'm a bit confused by this -> we can construct a starknet account struct, that has an ECDSA signing key, and perform calls with it?

Seems a bit misleading since the only signature that is "considered" by EOA is the one inside the RLP encoded signed tx (inside calldata), rather than in signed field.

But also, how does `let deployment_of_counter_evm_contract_result = eoa_starknet_account
        .execute(vec![Call {
            calldata: field_elements,
            to: not_used_eoa_account_fields,
            selector: not_used_eoa_account_fields,
        }])` sign the payload? Does it know a signature scheme?

Reply:
per our short call, will update this so its more clear and point to where people can get a clear view of how we handle eth payloads -> starknet transactions

```
    // This a made up signing key so we can reuse starknet-rs abstractions
    // to see the flow of how kakarot-rpc handles eth payloads -> starknet txns
    // see ./crates/core/src/client.rs::send_transaction 
    let signing_key = SigningKey::from_secret_scalar(FieldElement::ZERO);

    let eoa_starknet_account = SingleOwnerAccount::new(
        JsonRpcClient::new(HttpTransport::new(sequencer_url)),
        LocalWallet::from_signing_key(signing_key),
        eoa_account_starknet_address,
        chain_id::TESTNET,
    );

```